crates/holochain/CHANGELOG.md

```diff
@@ -7,6 +7,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 
+- Slightly simplify `GetLinksQuery` by removing a sub query. The sub query filtered `RegisterRemoveLink` ops by those that reference a `RegisterAddLink` op. That filter is not needed, because the logic then only filters out removed links from the result set if they reference a created link in the result set. Also it is very unlikely if not impossible that there are remove link actions in the database without the corresponding create link actions.
```

Comment:
> the logic then only filters out removed links from the result set if they reference a created link in the result set

I can't find this logic in `GetLinkDetailsQuery` -- it looks like the `render` function just naively returns the (create, Vec<delete>), without checking that the deletes correspond to the create. Can you point me to it?

Reply:
I wanna say it's this https://github.com/holochain/holochain/blob/develop/crates/holochain_state/src/query/link_details.rs#L63
but I don't see it being executed, except for the `Scratch` type. It works in tests. But maybe that's because the tests use `Scratch`? It doesn't matter, I'm not keen to dig into this any longer.